packages/web-components/src/calendar/calendar.ts

```diff
@@ -0,0 +1,813 @@
+        break;
+      }
+      case keyArrowLeft: {
+        event.preventDefault();
+        if (index === 0) {
+          this.handleSwitchSecondaryPanel('previous');
+          return true;
+        }
+        index = (index - 1) % this.secondaryPanelCells.length;
+        break;
+      }
+      case keyArrowDown: {
+        event.preventDefault();
+        if (index >= this.secondaryPanelCells.length - 4) {
+          this.handleSwitchSecondaryPanel('next');
+        }
+        index = (index + 4) % this.secondaryPanelCells.length;
+        break;
+      }
+      case keyArrowUp: {
+        event.preventDefault();
+        if (index < 4) {
+          this.handleSwitchSecondaryPanel('previous');
+        }
+        index = (index - 4 + this.secondaryPanelCells.length) % this.secondaryPanelCells.length;
+        break;
+      }
+      case keyHome: {
+        event.preventDefault();
+        // Update focus to the first cell on the panel
+        index = 0;
+        break;
+      }
+      case keyEnd: {
+        event.preventDefault();
+
+        // Update focus to the last cell on the panel
+        index = this.secondaryPanelCells.length - 1;
+        break;
+      }
+      default:
+        return true;
+    }
+
+    // Set focus on the cell corresponding to the updated index
+    currentCell.tabIndex = -1;
+    const focus = this.secondaryPanelCells[index] as HTMLElement;
+    focus.tabIndex = 0;
+    focus.focus();
+
+    // return true;
+  }
+
+  /**
+   * Handles keyboard events on the calendar control
+   * Includes navigating months using page up and page down keys
+   * @param event - Keyboard event
+   */
+  public handleControlKeydown(event: KeyboardEvent): boolean {
+    switch (event.key) {
+      case keyPageDown: {
+        event.preventDefault();
+        this.handleSwitchMonth(this.getMonthInfo().next.month, this.getMonthInfo().next.year);
+        break;
+      }
+      case keyPageUp: {
+        event.preventDefault();
+        this.handleSwitchMonth(this.getMonthInfo().previous.month, this.getMonthInfo().previous.year);
+        break;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Handles CSS animation classes for previous direction transitions on the primary panel
+   * @private
+   */
+  private prevMonthTransition() {
+    Updates.enqueue(() => {
+      const rows = this.shadowRoot && Array.from(this.shadowRoot?.querySelectorAll('.week'));
+
+      rows?.forEach(row => row.classList.add('animated-up'));
+
+      const firstTransitionRow = this.shadowRoot?.querySelector('.week-days')?.nextElementSibling as HTMLElement;
+
+      firstTransitionRow.classList.add('first-transition-row-animated');
+
+      //The timeout for the animation is set to the duration of the CSS animation as specified in the stylesheet
+      setTimeout(() => {
+        firstTransitionRow.classList.remove('first-transition-row-animated');
+        rows?.forEach(row => row.classList.remove('animated-up'));
+      }, 367);
```

Comment:
We might consider adding this as a named constant to maintain readability and maintenance. I saw @brianchristopherbrady called out the animation variable in the CSS. We should probably keep variable names consistent across CSS and TS as well.

Reply:
Gotcha. Where would I name these constants? Would these go in a different file?

Reply:
What if you put them in the file but mark them as private and not export them? Here's some constants you created elsewhere that are public and exported: https://github.com/microsoft/fluentui/pull/28585/files/24ee13a0ed015a80d1433b466d80f67325364aad#diff-c9b24cc36db628b77998616fcf4ada997a11bd3bd82c4b2509c00a85a1b2212cR10-R13

So instead of exporting them and making them public, declare them private and without the export prepended.